From f6b84bb10119d912ab293c6b49ba2399531c7aef Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Tue, 25 Feb 2025 16:26:24 +0100
Subject: [PATCH 873/954] fix(st-ddr): remove useless CONFIG_STM32MP25X_REVA
 flag

The SoC revision A is no more supported. Remove the corresponding flag.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 33bb629c133a95962d69336ac6b118ce611329fb fix(st-ddr): remove useless CONFIG_STM32MP25X_REVA flag]
Change-Id: Id867a6cd121de42010b7d179004f5ab1073e5bc9
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/439912
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 drivers/st/ddr/stm32mp2_ddr.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/st/ddr/stm32mp2_ddr.c b/drivers/st/ddr/stm32mp2_ddr.c
index 5d710df6f9..06f7e83eb6 100644
--- a/drivers/st/ddr/stm32mp2_ddr.c
+++ b/drivers/st/ddr/stm32mp2_ddr.c
@@ -70,9 +70,7 @@ static const struct stm32mp_ddr_reg_desc ddr_reg[DDRCTL_REG_REG_SIZE] = {
 	DDRCTL_REG_REG(init7, true),
 	DDRCTL_REG_REG(dimmctl, false),
 	DDRCTL_REG_REG(rankctl, true),
-#if !CONFIG_STM32MP25X_REVA
 	DDRCTL_REG_REG(rankctl1, true),
-#endif /* !CONFIG_STM32MP25X_REVA */
 	DDRCTL_REG_REG(zqctl0, true),
 	DDRCTL_REG_REG(zqctl1, false),
 	DDRCTL_REG_REG(zqctl2, false),
@@ -92,9 +90,7 @@ static const struct stm32mp_ddr_reg_desc ddr_reg[DDRCTL_REG_REG_SIZE] = {
 	DDRCTL_REG_REG(dbg1, false),
 	DDRCTL_REG_REG(dbgcmd, false),
 	DDRCTL_REG_REG(swctl, false), /* forced qd value */
-#if !CONFIG_STM32MP25X_REVA
 	DDRCTL_REG_REG(swctlstatic, false),
-#endif /* !CONFIG_STM32MP25X_REVA */
 	DDRCTL_REG_REG(poisoncfg, false),
 	DDRCTL_REG_REG(pccfg, false),
 };
@@ -146,10 +142,8 @@ static const struct stm32mp_ddr_reg_desc ddr_perf[DDRCTL_REG_PERF_SIZE] = {
 	DDRCTL_REG_PERF(perfhpr1, true),
 	DDRCTL_REG_PERF(perflpr1, true),
 	DDRCTL_REG_PERF(perfwr1, true),
-#if !CONFIG_STM32MP25X_REVA
 	DDRCTL_REG_PERF(sched3, false),
 	DDRCTL_REG_PERF(sched4, false),
-#endif /* !CONFIG_STM32MP25X_REVA */
 	DDRCTL_REG_PERF(pcfgr_0, false),
 	DDRCTL_REG_PERF(pcfgw_0, false),
 	DDRCTL_REG_PERF(pctrl_0, false),
-- 
2.53.0

